packages/map-template/src/components/OpeningHours/OpeningHours.jsx

```diff
@@ -0,0 +1,141 @@
+            t('Tuesday'),
+            t('Wednesday'),
+            t('Thursday'),
+            t('Friday'),
+            t('Saturday'),
+            t('Sunday')
+        ];
+
+        if (!isMondayFirstDayOfTheWeek) {
+            const sunday = days.pop();
+            days.unshift(sunday);
+        }
+        return days;
+    }, [t, isMondayFirstDayOfTheWeek]);
+
+    // Calculate adjusted current day index based on display order
+    const adjustedCurrentDay = useMemo(() => {
+        if (isMondayFirstDayOfTheWeek) {
+            // When Monday is first (0-6, Monday is 0)
+            return currentDay === 0 ? 6 : currentDay - 1;
+        } else {
+            // When Sunday is first (0-6, Sunday is 0)
+            return currentDay;
+        }
+    }, [currentDay, isMondayFirstDayOfTheWeek]);
+
+    const { standardOpeningHours } = openingHours || {};
+
+    const formatTime = (time) => {
+        const [hours, minutes] = time.split(':');
+        return new Date(0, 0, 0, hours, minutes).toLocaleTimeString([], {
+            hour: '2-digit',
+            minute: '2-digit',
+            hour12: isAmFormat
+        });
+    };
+
+    const getOpeningHoursForDay = (day) => {
+        const dayData = standardOpeningHours?.[day.toLowerCase()];
+        if (!dayData || dayData.closedAllDay) {
+            return {
+                text: t('Closed'),
+                isClosed: true
+            };
+        }
+        return {
+            text: `${formatTime(dayData.startTime)} - ${formatTime(dayData.endTime)}`,
+            isClosed: false
+        };
+    };
+
+    // Keep useCallback as this involves time calculations
+    const isCurrentlyOpen = useCallback((dayData) => {
+        if (!dayData || dayData.closedAllDay) return false;
+
+        const now = new Date();
+        const currentTime = now.getHours() * 60 + now.getMinutes();
+
+        const [startHours, startMinutes] = dayData.startTime.split(':');
+        const [endHours, endMinutes] = dayData.endTime.split(':');
+
+        const startTime = parseInt(startHours) * 60 + parseInt(startMinutes);
+        const endTime = parseInt(endHours) * 60 + parseInt(endMinutes);
+
+        return currentTime >= startTime && currentTime <= endTime;
+    }, []);
+
+    const getCurrentDayStatus = useCallback(() => {
+        const isOpen = isCurrentlyOpen(
+            standardOpeningHours?.[weekdays[adjustedCurrentDay].toLowerCase()]
+        );
+
+        return isOpen ? t('Open') : t('Closed');
+    }, [standardOpeningHours, weekdays, adjustedCurrentDay, isCurrentlyOpen, t]);
+
+    const text = getCurrentDayStatus();
+
+    return (
+        <div className="opening-hours">
+            <ul className="opening-hours__list">
+                <li className="opening-hours__list-item opening-hours__list-item--current" onClick={() => setIsExpanded(!isExpanded)}>
+                    <span className="opening-hours__time">
+                        <div className="opening-hours__icon-wrapper">
```

Comment:
Not a fan of having block level elements (div) inside inline element (span). Can we use a span here as well for semantic reasons? Or change the parent span to a div?